cmd/server/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/redhatinsights/insights-operator-conditional-gathering/pkg/service"
+	"github.com/redhatinsights/insights-operator-conditional-gathering/pkg/service/transport"
+	"golang.org/x/sync/errgroup"
+)
+
+func main() {
+	var (
+		// Flags
+		httpAddr  = flag.String("httpAddr", ":8080", "http listen address")
+		rulesPath = flag.String("rulesPath", "./rules", "the path where to find the rules files")
+
+		httpServer *http.Server
+	)
+
+	flag.Parse()
```

Comment:
No problem to change it to config and env variables. I was considering to do it, just used flags 'cause it was very simple implementation.

Now, a really honest question: why not to use flags? could you give me some context?

Reply:
Sure. Basically it is not our decision. The thing is that configuration options are stored as "secrets" in OpenShift and there's mapping between secret name and environment variable name. So to make these things really configurable (needed for example for stage environment, where HTTP port might be different) it's better to read config from variables too. It can be done via `viper` library (pretty nice one).

Reply:
Gotcha! Thank you.

Reply:
Done. It is now using viper.